src/Board.py

```diff
@@ -1,23 +1,27 @@
+from typing import List, no_type_check
+
+from termcolor import colored
+
 from src.Bishop import Bishop
 from src.Coordinate import Coordinate as C
 from src.King import King
 from src.Knight import Knight
 from src.Move import Move
 from src.Pawn import Pawn
+from src.Piece import Piece
 from src.Queen import Queen
 from src.Rook import Rook
-from termcolor import colored
 
 WHITE = True
 BLACK = False
 
 
 class Board:
 
-    def __init__(self, mateInOne=False, castleBoard=False,
-                 passant=False, promotion=False):
-        self.pieces = []
-        self.history = []
+    def __init__(self, mateInOne: bool = False, castleBoard: bool = False,
+                 passant: bool = False, promotion: bool = False):
+        self.pieces: List[Piece] = []
+        self.history: List[List] = []  # type: ignore[type-arg]  # TODO: make 'history' contain only Move, not None
```

Comment:
Eventually I'm blind, but i don't see when history would contain a `None`.

Reply:
> `history` is declared to be `list[list]` which is automatically translated to `list[list[Any]]`. So I assume mypy isn't happy for the following reasons:
> - Line 166: mypy can't infer the return type because the element which is accessed (`history[-1][0]` where `[0]` is the important part) is declared to be `Any`
> - Line 171: same as in line 166. mypy can't infer what type `history[-1][0].piece` is because `history[-1][0]` is already `Any`.

In connection with this it would be an idea to change the type of history to be `list[tuple[Piece, Optional[Piece]]]`. And all the places where this type is expected have to be changed too.

Reply:
This makes sense, I think you meant `list[tuple[Move, Optional[Piece]]]` though. See if this makes sense now.

Reply:
> I think you meant list[tuple[Move, Optional[Piece]]] though

Yeah, you're right with that.